azext_edge/edge/util/x509.py

```diff
@@ -78,11 +81,25 @@ def generate_self_signed_cert(valid_days: int = DEFAULT_VALID_DAYS) -> Tuple[byt
     return (cert.public_bytes(serialization.Encoding.PEM), key_bytes)
 
 
-def decode_der_certificate(der_data: bytes) -> Optional[x509.Certificate]:
-    # Decodes a DER-encoded X.509 certificate.
+def decode_x509_files(
+    cert_data: bytes,
+    content_format: str,
+    extension: str
+) -> Optional[List[Union[x509.Certificate, x509.CertificateRevocationList]]]:
+    # Decodes an X.509 file from PEM or DER format.
+    certs: List = []
     try:
-        cert = x509.load_der_x509_certificate(der_data, default_backend())
-        return cert
+        if content_format == X509FileExtension.PEM.name:
+            # PEM format
+            certs = x509.load_pem_x509_certificates(cert_data)
```

Comment:
this doesn't need default_backend()?

Reply:
nope
<img width="208" alt="image" src="https://github.com/user-attachments/assets/5d9bb618-8ec1-4444-84c1-86d27ee983be" />